configurd.go

```diff
@@ -58,12 +66,16 @@ func relativePath(path, basePath string) string {
 	return strings.Replace(path, basePath+"/", "", -1)
 }
 
-func New(log Logger, basePath, defaultTag string) (Configurd, error) {
+func New(config Config) (Configurd, error) {
```

Comment:
Since Configurd is going to need all of the fields in Config, might be a better idea to instantiate Config inside New instead of outside.
So:
```
func New(log Logger, basePath, defaultTag string) (Configurd, error) {
...
	c := Configurd{
		config: Config{...},
	}
```

Reply:
The standard way would be to pass down the logger so that it can be configured